### PR TITLE
[5.0] Update phpseclib to 3.0.34

### PR DESCRIPTION
**Pull request by SniperSister**
### Summary of Changes
Update phpseclib to 3.0.34 to fix https://nvd.nist.gov/vuln/detail/CVE-2023-49316.

### Testing Instructions
Code review.

### CODE REVIEW

**Comment by richard67** [comment]
This will also fix issue #42142 . See also my PR #42190 for that issue, which I've just closed in favour of this one here.

File: composer.lock [hunk 1]
```diff
@@ -9996,5 +9996,5 @@
     "platform-overrides": {
         "php": "8.1.0"
     },
-    "plugin-api-version": "2.6.0"
+    "plugin-api-version": "2.3.0"
```
Comment (richard67) [inline composer.lock]: Is this change intended, or does it just come from using the same version as for PR #42470 ?

Comment (richard67) [reply composer.lock]: Anyway I think it doesn't really matter, just am curious.

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 674f372068e47927d0b2bd3c7b1f67a6a741b81d<br /><br />Tested by code review + verified that the URL is correct.

There is an additional change from `"plugin-api-version": "2.6.0"` to `"plugin-api-version": "2.3.0"`, but that doesn't really matter, so I'm ok with it as it is, but would also be ok with reverting that change.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42469">issues.joomla.org/tracker/joomla-cms/42469</a>.</sub>

**Comment by richard67** [comment]
@SniperSister Meanwhile there is a new release 3.0.35 available. Changelog see https://github.com/phpseclib/phpseclib/releases/tag/3.0.35 . Would it make sense to update this PR to that release?

**Comment by bembelimen** [comment]
Thx

**Pull request merged**